File: app/src/main/java/com/example/imagemachine/feature/main/presenter/MainPresenter.java
package com.example.imagemachine.feature.main.presenter;

import com.example.imagemachine.feature.base.presenter.BasePresenter;
import com.example.imagemachine.feature.main.view.IMainView;

public class MainPresenter<V extends IMainView> extends BasePresenter<V> implements IMainPresenter {

    public MainPresenter(V view) {
        super(view);
    }

    @Override
    public void viewDidAttach() {
        super.viewDidAttach();
    }

    @Override
    public void viewDidDetach() {
        super.viewDidDetach();
    }
}

File: app/src/main/java/com/example/imagemachine/feature/main/presenter/MainPresenter.kt
package com.example.imagemachine.feature.main.presenter

import com.example.imagemachine.feature.base.presenter.BasePresenter
import com.example.imagemachine.feature.main.view.IMainView

internal class MainPresenter<V: IMainView>(view: V): BasePresenter<V>(view), IMainPresenter<V> {
}